internal/api/sms_provider/sms_provider.go

```diff
@@ -27,6 +29,8 @@ func init() {
 
 type SmsProvider interface {
 	SendMessage(phone, message, channel string) error
+	SendVerification(phone, channel string) error
+	VerifyOTP(phone, otp string) error
```

Comment:
is there a way to not include this in the interface since it only applies to twilio? we could also create a default `SendVerification` method for the `SmsProvider` which can be overridden by twilio's `SendVerification` method - that should help to cut down some LOC in `internal/api/phone.go`